pymilo/utils/util.py

```diff
@@ -58,3 +58,34 @@ def check_str_in_iterable(field, content):
         return False
     else:
         return field in content
+
+
+def get_homogeneous_type(seq):
+    """
+    Check if the given sequence's inner items have the same type or not and if they do, return the associated type.
+
+    :param seq: given sequence
+    :type seq: sequence
+
+    :return: Tuple of (True, inner_type) or (False, None)
+    """
+    iseq = iter(seq)
+    first_type = type(next(iseq))
+    return (
+        True,
+        first_type) if all(
+        (type(x) is first_type) for x in iseq) else (
+            False,
+        None)
```

Comment:
This function is a little bit odd. It actually both validates and return the first objects type.
I suggest changing it to `_check_homogeneity` and do the validation part only (boolean return type or an exception raiser. Do the rest by using `type(seq[0])` in the main code.

Reply:
Hmm, why? it checks that the given array's items have the same type and returns the type too.
I thought putting both logics in the same function might be more cohesive.

Reply:
Let's see what @sepandhaghighi says about it.